### PR TITLE
Change functions name for partial support of cuDNN v4

### LINKED ISSUE #835: chainer not compatible with cudnn-7.0-linux-x64-v4.0
Running `python setup.py build` in master branch of e99cf1594c5bc70825bd156f6684a37ef6ffe0c5 results in following error, when `cudnn-7.0-linux-x64-v4.0` is in path.

The same commit successfully builds with `cudnn-6.5-linux-x64-v2` .

````
cupy/cuda/cudnn.cpp:4926:398: error: cannot convert ‘void*’ to ‘cudnnConvolutionBwdFilterAlgo_t’ for argument ‘8’ to ‘cudnnStatus_t cudnnConvolutionBackwardFilter(cudnnHandle_t, const void*, cudnnTensorDescriptor_t, const void*, cudnnTensorDescriptor_t, const void*, cudnnConvolutionDescriptor_t, cudnnConvolutionBwdFilterAlgo_t, void*, size_t, const void*, cudnnFilterDescriptor_t, void*)’

   __pyx_v_status = cudnnConvolutionBackwardFilter(((cudnnHandle_t)__pyx_v_handle), ((void *)__pyx_v_alpha), ((cudnnTensorDescriptor_t)__pyx_v_srcDesc), ((void *)__pyx_v_srcData), ((cudnnTensorDescriptor_t)__pyx_v_diffDesc), ((void *)__pyx_v_diffData), ((cudnnConvolutionDescriptor_t)__pyx_v_convDesc), ((void *)__pyx_v_beta), ((cudnnFilterDescriptor_t)__pyx_v_gradDesc), ((void *)__pyx_v_gradData));

                                                                                                                                                                                                                                                                                                                                                                                                              ^

cupy/cuda/cudnn.cpp: In function ‘PyObject* __pyx_f_4cupy_4cuda_5cudnn_convolutionBackwardData(size_t, size_t, size_t, size_t, size_t, size_t, size_t, size_t, size_t, size_t, int)’:

cupy/cuda/cudnn.cpp:5145:402: error: cannot convert ‘void*’ to ‘cudnnConvolutionBwdDataAlgo_t’ for argument ‘8’ to ‘cudnnStatus_t cudnnConvolutionBackwardData(cudnnHandle_t, const void*, cudnnFilterDescriptor_t, const void*, cudnnTensorDescriptor_t, const void*, cudnnConvolutionDescriptor_t, cudnnConvolutionBwdDataAlgo_t, void*, size_t, const void*, cudnnTensorDescriptor_t, void*)’

   __pyx_v_status = cudnnConvolutionBackwardData(((cudnnHandle_t)__pyx_v_handle), ((void *)__pyx_v_alpha), ((cudnnFilterDescriptor_t)__pyx_v_filterDesc), ((void *)__pyx_v_filterData), ((cudnnTensorDescriptor_t)__pyx_v_diffDesc), ((void *)__pyx_v_diffData), ((cudnnConvolutionDescriptor_t)__pyx_v_convDesc), ((void *)__pyx_v_beta), ((cudnnTensorDescriptor_t)__pyx_v_gradDesc), ((void *)__pyx_v_gradData));

                                                                                                                                                                                                                                                                                                                                                                                                                  ^

error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /usr/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_ubuntu/chainer/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-KqXMta-record/install-record.txt --single-version-externally-managed --compile --user failed with error code 1 in /tmp/pip_build_ubuntu/chainer
Traceback (most recent call last):
  File "/usr/bin/pip", line 9, in <module>
    load_entry_point('pip==1.5.4', 'console_scripts', 'pip')()
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 235, in main
    return command.main(cmd_args)
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 161, in main
    text = '\n'.join(complete_log)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 42: ordinal not in range(128)
````

### PR DESCRIPTION
**Pull request by okuta**
This PR fix #835.

### CODE REVIEW

**Comment by okuta** [comment]
This pull request is related to #756.

File: cupy/cuda/cupy_cudnn.h [hunk 1]
```diff
@@ -8,6 +8,16 @@
 #ifndef CUPY_NO_CUDA
 #include <cudnn.h>
 
+#if CUDNN_VERSION < 4000
```
Comment (unnonouno) [inline cupy/cuda/cupy_cudnn.h]: Please write a comment about these defines.

File: cupy/cuda/cupy_cudnn.h [hunk 2]
```diff
@@ -8,6 +8,16 @@
...
+
+#define cudnnSetConvolutionNdDescriptor_v2 cudnnSetConvolutionNdDescriptor
+#define cudnnGetConvolutionNdDescriptor_v2 cudnnGetConvolutionNdDescriptor
+#define cudnnAddTensor_v2 cudnnAddTensor
+#define cudnnConvolutionBackwardFilter_v2 cudnnConvolutionBackwardFilter
+#define cudnnConvolutionBackwardData_v2 cudnnConvolutionBackwardData
+
+#endif
```
Comment (unnonouno) [inline cupy/cuda/cupy_cudnn.h]: Write `// #if CUDNN_VERSION < 4000`

**Comment by unnonouno** [comment]
LGTM!

**Comment by unnonouno** [comment]
cuDNN v4 is not released yet. So, please remove v4 from README.

**Pull request merged**